tools/backups2datalad.py

```diff
@@ -462,52 +451,81 @@ def mkrelease(
         commitish: Optional[str] = None,
     ) -> None:
         # `dandiset` must have its version set to the published version
+        repo: Path = ds.pathobj
+        remote_assets = list(dandiset.get_assets())
+
         def git(*args: str, **kwargs: Any) -> None:
             log.debug("Running: git %s", " ".join(shlex.quote(str(a)) for a in args))
             subprocess.run(["git", *args], cwd=repo, check=True, **kwargs)
 
-        repo: Path = ds.pathobj
+        def commit_has_assets(commit_hash: str) -> bool:
+            repo_assets = json.loads(
+                readcmd("git", "show", f"{commit_hash}:.dandi/assets.json", cwd=repo)
+            )
+            return (not remote_assets and not repo_assets) or (
+                repo_assets
+                and isinstance(repo_assets[0], dict)
+                and "asset_id" in repo_assets[0]
+                and assets_eq(remote_assets, repo_assets)
+            )
+
+        candidates: List[str]
         if commitish is None:
-            commitish = readcmd(
-                "git",
-                "rev-list",
-                f"--before={dandiset.version.created}",
-                "-n1",
-                "HEAD",
-                cwd=repo,
+            candidates = []
+            candidates.append(
+                readcmd(
+                    "git",
+                    "rev-list",
+                    f"--before={dandiset.version.created}",
+                    "-n1",
+                    "HEAD",
+                    cwd=repo,
+                )
             )
-        remote_assets = list(dandiset.get_assets())
-        repo_assets = json.loads(
-            readcmd("git", "show", f"{commitish}:.dandi/assets.json", cwd=repo)
+            if (
+                cmt := readcmd(
+                    "git",
+                    "rev-list",
+                    f"--after={dandiset.version.created}",
+                    "-n1",
+                    "HEAD",
+                    cwd=repo,
+                )
+            ) != "":
```

Comment:
I would have preferred some more direct of "get direct children  of `committish` here.
also ain't with `--after={dandiset.version.created}` we might miss a commit which has exactly that stamp... I think it is unlikely to happen and likely the only commit with that date will be the tagged release commit (which will end up on the HEAD after #69 merged), which we should ignore here anyways.
So I guess we can proceed with this as is for now.

Reply:
Experimentation indicates that `--before` and `--after` will select a commit with exactly the given date & time.